### DISCUSSION TITLE
Logarithmic spectrum (dB scale)

### DISCUSSION THREAD

**Discussion by s57uuu** [ra3xdh/qucs_s#335] [Q&A] [answered]
Hello!

I am new to qucs-s. I have used qucs for AC and S parameter simulation before, mostly for RF stuff, like filter design etc.

Now I am trying transient simulation with qucs-s and ngspice. For start I tried a simple oscillator. The first (attached osc1_old.sch file) version had a "cold start". Second and third harmonics can be barely seen on the spectrum, just a couple of pixels. A log graph would be better in showing them, so I added eq2. But it looks (in the log) like qucs is trying to evaluate eq2 before doing the FFT, so no results.

Next I added a "kickstart", by adding initial voltages to capacitors, so I could do a shorter simulation (attached file osc1.sch). Now I do not see it even trying to evaluate eq2 in the log? Also no trace of eq2 in the .cir file.

Marko Cebokli

[osc1_old.sch.txt](https://github.com/ra3xdh/qucs_s/files/13062507/osc1_old.sch.txt)
[osc1.sch.txt](https://github.com/ra3xdh/qucs_s/files/13062508/osc1.sch.txt)

[osc1.dpl.txt](https://github.com/ra3xdh/qucs_s/files/13062511/osc1.dpl.txt)

**Comment by ra3xdh** [comment, marked as answer]
> But it looks (in the log) like qucs is trying to evaluate eq2 before doing the FFT, so no results.

You should specify the `Simulation=fft` instead of `ac` in the equation properties. FFT and AC are different simulations. After applying these corrections everything works as expected. 

> Now I do not see it even trying to evaluate eq2 in the log?

This question was asked and answered several times before. See the recent discussion #317 and #226. And there exists more old issues referencing the IC setting. You should set `InitialDC=no` in the transient simulation options (the second tab of the dialog window). This will apply the IC to the schematic. I don't know how to make this option better visible for users and prevent asking of this question in the future. Any suggestion os welcome. Obvious answer may be to add the description of TRAN+IC simulation in the documentation. 

Attached screenshot shows the fixed schematic:
![image](https://github.com/ra3xdh/qucs_s/assets/4920080/68f8c299-d3c7-4b61-bb50-374e1604c0ae)

**Comment by s57uuu** [comment]
Thank you very much for the quick answer!   In fact, I did find out about the initialDC thing before (it is already in my second file in the previous post).

If I am interested just in the steady state spectrum (without the switch-on transient), can I somehow tell the FFT analyzer to only take samples from 1E-7 to 2e-7 seconds? (or any other interval)

**Comment by ra3xdh** [comment]
> can I somehow tell the FFT analyzer to only take samples from 1E-7 to 2e-7 seconds?

No, there exists no way to set the start time of the FFT. I may consider to add this option in the future. As a workaround you have to construct the simulation using the nutmeg script simulation type. An example is shown here:

![image](https://github.com/ra3xdh/qucs_s/assets/4920080/b16e0d27-0b11-4473-ac70-b6b876e34e03)

**Reply by s57uuu** [reply]
OK, thanks!

**Comment by ra3xdh** [comment]
I have created a new task for this request. See #336 The tstart option may appear in the next release.
